docs/content/sips/xxx-trigger-executors.md

```diff
@@ -0,0 +1,68 @@
+title = "SIP xxx - Trigger Executors"
+template = "main"
+
+---
+
+Summary: A new architecture for Spin trigger executors.
+
+Owner: lann.martin@fermyon.com
+
+Created: April 29, 2022
+
+<!-- Updated: April 4, 2022 -->
+
+## Background
+
+Spin currently includes built-in implementations for two trigger types: HTTP, and Redis (PubSub).
+We want to allow for additional trigger types in the future, whether they are "internal" types developed and
+supported within the Spin codebase, or "external" types developed by the broader community.
+
+While it is currently possible to build external trigger types - as in the
+[TimerTrigger example](https://github.com/fermyon/spin/blob/main/examples/spin-timer/src/main.rs) - it requires replicating much of the
+functionality provided by the `spin up` command.
+
+## Proposal
+
+### Trigger types are implemented as "trigger executors"
+
+These executors are regular subcommands/binaries, but are not normally executed directly by
+users.
+
+* "Internal" triggers use `spin` subcommands, e.g. `spin trigger http`
+* "External" triggers use external binaries, e.g. `spin-trigger-timer`
+* Trigger executors share common code, including common CLI options in a `spin-trigger` crate
+
+### The `spin up` command changes
+
+It remains the primary entrypoint for starting a Spin app, but its behavior is changed to:
+
+1. Load the application manifest
+1. Find a trigger executor for the application (e.g. check the `PATH` for `"spin-trigger-${trigger.type}"`)
+1. Set some environment variables:
+   * `SPIN_MANIFEST_URL`: a URL pointing to the application manifest (e.g. `file:///path/to/spin.toml` or `bindle+https://bindle-server?id=bindle-id`)
```

Comment:
The `bundle+https` URL here is just one option to support future-compatibility with other loaders.